### DISCUSSION TITLE
`forge-alist`: support `ssh.github.com`

### DISCUSSION THREAD

**Discussion by gekoke** [magit/forge#708] [Ideas]
GitHub remotes can be configured using the `ssh.github.com` domain. When configuring a GitHub remote using SSH authentication behind a firewall that messed with port 22 using [the steps outlined here](https://docs.github.com/en/authentication/troubleshooting-ssh/using-ssh-over-the-https-port), `forge` was unable to determine the correct forge.

I remidiated this with `(add-to-list 'forge-alist '("ssh.github.com" "api.github.com" "github.com" forge-github-repository))` in my config.

Perhaps this value should exist by default in the alist?

**Comment by tarsius** [comment]
Thanks for making me aware of this potential complication!

I'm undecided whether to add an entry to `forge-alist` and/or a note to the documentation somewhere. It has been a struggle to find the right balance in the past. Before I added additional information about complications affecting only a small subset of users, people told me that there was not enough documentation. After I added such information, people told me there was too much irrelevant information and they started asking for simple tutorials, and I noticed that many stopped reading even the parts that were relevant for everyone.

I'll let this marinate a bit, before deciding what to do.

**Reply by gekoke** [reply]
Thank you for your consideration! 

The fix in this case was, admittedly, a relatively easy and permanent one.

**Reply by tarsius** [reply]
I've just added an entry to `forge-alist`'s default value, in 99f690aa181b23c76988fff8c55cd64f1dc60af6.
